### DISCUSSION TITLE
Link-Local IP Address

### DISCUSSION THREAD

**Discussion by skcole67** [sstaub/Ethernet3#48] [Q&A]
Hi

Do you have any insight into how I may generate a local-network unique link-local address?

I want to make the library automatically return a network-unique random Link-Local address if DHCP fails. And maybe, if not too difficult, confirm its continued uniqueness during operation. Perhaps supporting a reboot if the uniqueness is lost. Maybe.

Still at the initial stages of thinking about this.

Thanks for a great base library!

SKC

**Comment by sstaub** [comment]
The only way for generating unique IP addresses is a DHCP server, they are build mostly in Layer3 switches like Cisco SG-350 or WIFI Access routers.

**Reply by skcole67** [reply]
I have been reading RFC3927 - Dynamic Configuration of IPv4Link-Local Addresses. And it means sending/receiving ARP Request packets. Can this be done with the W5500?! It suggests ARP sending is possible but I am not sure if it is buried in the chip or in library code. I notice the Force-ARP register. Can I send and control ARP packets or does W5500 isolate you too much from such low-level features?

**Comment by sstaub** [comment]
There is only a ARP force register but I don't know which mode is standard. But there is no ARP control by the user, this is done internal.
Force ARP is intended to update the ARP table, nothing else. Sorry you need a DHCP server.

**Reply by skcole67** [reply]
Is this not even possible?
[http://www.zeroconf.org/QDIPv4LL.html](http://www.zeroconf.org/QDIPv4LL.html)

**Comment by sstaub** [comment]
There is a mDNS library for Ethernet3 https://github.com/michael71/EthernetBonjour3, but no explicit "zeroconfig" library.
Zeroconfig is normally used when there are two or more 'same' static IP addresses are found in the network.
